deployment/v3/utils/prop_migrator/README.md

```diff
@@ -0,0 +1,137 @@
+
+-----
+# Property File Updater
+
+The Property File Updater is a script that automates the process of updating property files based on information provided in knowledge base CSV files. It allows you to manage and prioritize property values, handle different versions of property files, and generate a manual configuration file for further customization.
+
+This script updates property files in a given directory based on information provided in knowledge base CSV files. It performs the following actions:
+
+1. Reads a CSV file (`different_properties.csv`) containing information about different property values.
+2. Reads two CSV files (`old-value-takes-priority.csv` and `latest-Value-takes-priority.csv`) to prioritize property values.
+3. Compares non-matching combinations with `latest-Value-takes-priority.csv` and identifies similar latest values.
+4. Creates a `manual-configuration.csv` file for further manual configuration.
+5. Compares combinations from `different_properties.csv` with `old-value-takes-priority.csv` and `latest-Value-takes-priority.csv` to find unresolved combinations.
+6. Updates the `manual-configuration.csv` file with unresolved combinations.
+7. Reads `old_file_only.csv` and `intentionally_removed_in_lts.csv` files and compares the combinations.
+8. Updates the `manual-configuration.csv` file with the new combinations.
+
+## Prerequisites
+
+
+- Python 3.x
+- Required Python packages: `csv`
+
+## Usage
+
+1. Ensure that you have the required CSV files and the latest property files in the correct directories.
+2. Open a terminal and navigate to the directory containing the script.
+3. Run the script with the latest repository branch as a command-line argument:
+
+   ```bash
+   python property_file_updater.py <latest_repo_branch>
+
+4. The script will update the property files and generate a log file (logs.txt) with the details of the update process.
+5. It will also create or update the manual-configuration.csv file with unresolved combinations for further manual configuration.
+
+## Output
+
+The Property File Updater generates the following output:
+
+### Updated Property Files
+
+This section lists the property files that were updated during the process. It includes the names of the files and the keys that were modified.
+
+### Logs
+
+The script generates a log file (`logs.txt`) that provides detailed information about the update process. It includes information about matching combinations, unresolved combinations, similar latest values, and more. The log file helps you understand the changes made to the property files and allows you to review the update process.
+
+### Manual Configuration CSV
+
+The script creates or updates the `manual-configuration.csv` file. This file includes unresolved combinations and combinations that require further manual configuration. You can use this file to manually update the property files based on the required configuration actions mentioned in the file.
```

Comment:
Need more details on the structure of csv as a example like it is shown in the overview file and what all are the different "Configuration Action" can be seen in this output file and what a end user is suppose to understand and do from each of them